src/transaction/log_recovery.c

```diff
@@ -2911,6 +2911,23 @@ log_recovery_analysis_internal (THREAD_ENTRY * thread_p, INT64 * num_redo_log_re
 	  break;
 	}
 
+      if (is_passive_transaction_server ())
```

Comment:
What if the current log record is not an MVCC record?

This information - the largest encountered mvccid - can be collected also in the following functions:
- log_rv_analysis_record
  - log_rv_analysis_record_on_tran_server
    - log_rv_analysis_mvcc_undo_redo (need to pass `log_recovery_context` instance as an argument, the rest of the information is there)
    - log_rv_analysis_assigned_mvccid (same, pass `log_recovery_contenxt` as argument)
It is more generic as already needed information is there: proper MVCCID log record, tdes, mvccid value.

I ask to move the collection of the "largest mvccid" because this game me an idea of improving the missing mvccid's handling (see comment before `log_Gl.mvcc_table.complete_mvccids_if_still_active`):
- the `log_recovery_context` can be extended to also take the set of "missing mvccids" and substract from them along the way while executing the log recovery analysis
- thus it would minimize even more the uncertainty and would make the logic easier to debug if the assumption described in the aforementioned comment is ever broken
But this is for another time.

Reply:
> What if the current log record is not an MVCC record?

The mvccid of the transaction that left the log will be used. If there is no mvcc operation within that transaction, then `MVCCID_NULL` will be used. 

> This information - the largest encountered mvccid - can be collected also in the following functions:
> 
> * --> `log_rv_analysis_record`
>   
>   * --> `log_rv_analysis_record_on_tran_server`
>     
>     * --> `log_rv_analysis_mvcc_undo_redo` (need to pass `log_recovery_context` instance as an argument, the rest of the information is there)
>     * --> `log_rv_analysis_assigned_mvccid` (same, pass `log_recovery_contenxt` as argument)
>       It is more generic as already needed information is there: proper MVCCID log record, tdes, mvccid value.

I am aware of the detailed function for parsing mvcc records (`log_rv_analysis_assigned_mvccid`, `log_rv_analysis_mvcc_undo_redo`). However, updating the variables used only for the passive transaction server seems to be out of the scope of the above functions, so I modified it as in the current PR.
 
> I ask to move the collection of the "largest mvccid" because this game me an idea of improving the missing mvccid's handling (see comment before `log_Gl.mvcc_table.complete_mvccids_if_still_active`):
> 
> * the `log_recovery_context` can be extended to also take the set of "missing mvccids" and substract from them along the way while executing the log recovery analysis
> * thus it would minimize even more the uncertainty and would make the logic easier to debug if the assumption described in the aforementioned comment is ever broken
>   But this is for another time.

This is also a fact recognized while analyzing recovery from PTS. If we can remove mvccid from `missing mvccids` during analysis, we can filter out `active mvccid` from the missing mvccid set. But, as you said, this is for another time. I will make another issue about this one.

Summary : 
I'm not sure if it would be better to move the update of largest mvccid inside those function ((`log_rv_analysis_assigned_mvccid`, `log_rv_analysis_mvcc_undo_redo`) . However, I think it will be less expensive than finding log_tdes for all log records and comparing mvccid.

I will fix it. thanks.

Reply:
The comparison will still be needed.
Howevever it will be one `logtb_find_tran_index` and one `LOG_FIND_TDES` less for each log record.

Reply:
Thanks. I fixed.